unit_tests/smoke_static.cpp

```diff
@@ -144,5 +142,5 @@ main(
 
     // Finalize MPI and return.
     MPI_Finalize();
-    return !status;
+    return 0;
```

Comment:
It looks like the goal of this test just to run without failing? There is nothing checked, like in `test_Matrix.cpp` where `EXPECT*` functions are called. Is this the intention?

Reply:
Yes this test was added to the CI only intending to be a quick check if an error occurred with `endSamples()`/`writeSnapshot()`. The issue that caused this problem has since been fixed, but the only test I could find that uses these functions is in `test_HDFDatabase.cpp` which is restricted to parallel HDF5 builds only.
In the future, we should provide a serial test in `test_HDFDatabase.cpp` and remove this test (and other outdated/unused tests).